### ISSUE TITLE
Seg Fault upon Connecting Two Vehicles with Tow Cable 

### ISSUE THREAD

**Issue by ShnitzelX2** [CleverRaven/Cataclysm-DDA#71412]
### Describe the bug

Connecting two vehicles together with a tow cable, removing the cable, and then again connecting them causes a segmentation fault. Also, whether attaching the cable to a vehicle works or not seems inconsistent.

### Attach save file

[Mishicot-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/14130859/Mishicot-trimmed.tar.gz)

The save file isn't necessary, you should be able to do this in a new world. 

### Steps to reproduce

1. Spawn two sedans, as oriented in the attached screenshot.
2. Attach a heavy-duty tow cable first to the headlight of the north sedan (highlighted in red on the screenshot) for *towing*. (NOTE: often this will be rejected by the game with "You can't attach a tow-line to a racked part", seemingly randomly for either vehicle but often for both. I'm not sure if this is also a bug. Try this process a couple times.)
3. Attach the other end to the front of the south sedan (highlighted in blue on the screenshot).
4. Remove the tow cable from the north sedan.
5. Repeat step 2; this crashes the game.

### Expected behavior

I expected either the cars to connect or to be rejected for connecting the cars in the wrong spot on the vehicle.

### Screenshots

![towcable_bug](https://github.com/CleverRaven/Cataclysm-DDA/assets/65314588/47b8a0db-b045-41a8-af29-c0ee7b261941)

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.3930 (22H2)
- Game Version: fd88c9f [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth]
]
/

### Additional context

[crash.log](https://github.com/CleverRaven/Cataclysm-DDA/files/14130879/crash.log)
[debug.log](https://github.com/CleverRaven/Cataclysm-DDA/files/14130880/debug.log)

While this bug is reproduced here with spawned vehicles, it did occur naturally in a different world of mine the first time I tried to connect two vehicles with a tow cable. Since it's easily reproducible, I didn't bother including that save file.

**Comment by PatrikLundell** [comment]
/confirmed

- I completely failed to connect the tow cable to the northern vehicle in the westmost pair, trying the tail light several times, then each of the other rear tiles, all with the racked part message.
- I then gave up on that and tried to attach the cable on the headlight of the southern vehicle (as the towee), at which the game CTD:ed.
- Starting the save again I tried to attach to the back door of the northern vehicle as the tower, again resulting in a CTD.

- Started my own save (or, rather, testing offshoot of it), spawned a tow cable (too lazy to locate one) and tried to attach it as a tower to the rear of one of my vehicles, resulting in a CTD.

Thus, there are two errors here:
- Erroneous racked part message; and
- CTD.
